spec/models/question_spec.rb

```diff
@@ -6,10 +6,66 @@
     it { should belong_to(:category) }
     it { should have_many(:answers) }
     it { should have_many(:comments) }
+    it { should have_many(:upvotes) }
+    it { should have_many(:downvotes) }
   end
 
   context "validations" do
     it { should validate_presence_of(:title) }
     it { should validate_presence_of(:body) }
   end
+
+  describe ".answer_count" do
+    it "returns the question's answer count" do
+      user = Fabricate(:user, name: "we the best")
+      question = Fabricate(:question, title: "Why do they try to stop us?", user: user)
+      answer_1 = question.answers.create(body: "blessup", user: user)
+      answer_2 = question.answers.create(body: "we the best", user: user)
+      answer_3 = question.answers.create(body: "$$$", user: user)
+
+      expect(question.answer_count).to eq(3)
+    end
+  end
+
+  describe ".current_user_upvote_correction" do
+    it "checks if a user's upvote exists on a question and deletes it" do
+      user_1 = Fabricate(:user)
+      user_2 = Fabricate(:user)
+      user_3 = Fabricate(:user)
+      user_4 = Fabricate(:user)
+
+      question = Fabricate(:question, title: "Why are we the best?", body: "Bless Up", user: user_1)
+
+      upvote_1 = question.upvotes.create(creator: user_2.id, user_id: user_1.id)
+      upvote_2 = question.upvotes.create(creator: user_3.id, user_id: user_1.id)
+      upvote_3 = question.upvotes.create(creator: user_4.id, user_id: user_1.id)
+
```

Comment:
Good work figuring everything out to produce good tests. Thanks for the work you did @maxglassie